import "maidsafe/dht/kademlia.proto";

package maidsafe.dht.protobuf;

message PingRequest {
  required Contact sender = 1;
  required bytes ping = 2;
  required bytes id = 3;
}

message PingResponse {
  required bytes echo = 1;
  required bytes id = 2;
}

message FindValueRequest {
  required Contact sender = 1;
  required bytes key = 2;
  optional int32 num_nodes_requested = 3;
  required bytes id = 4;
}

message FindValueResponse {
  required bool result = 1;
  repeated Contact closest_nodes = 2;
  optional Contact alternative_value_holder = 3;
  required bytes id = 4;
}

message FindNodesRequest {
  required Contact sender = 1;
  required bytes key = 2;
  optional int32 num_nodes_requested = 3;
  required bytes id = 4;
}

message FindNodesResponse {
  required bool result = 1;
  repeated Contact closest_nodes = 2;
  required bytes id = 3;
}

